src/domain/collaboration/aspect/aspect.resolver.subscriptions.ts

```diff
@@ -28,60 +29,52 @@ export class AspectResolverSubscriptions {
   @UseGuards(GraphqlGuard)
-  @Subscription(() => AspectCommentsMessageReceived, {
-    description: 'Receive new comment on Aspect',
-    async resolve(
-      this: AspectResolverSubscriptions,
-      value: AspectCommentsMessageReceived,
-      _: any,
-      context: any
-    ): Promise<AspectCommentsMessageReceived> {
-      const agentInfo = context.req?.user;
-      const logMsgPrefix = `[User (${agentInfo.email}) Aspect comments] - `;
-      this.logger.verbose?.(
-        `${logMsgPrefix} sending out new message event for Aspect: ${value.aspectID} `,
-        LogContext.SUBSCRIPTIONS
-      );
-      return value;
-    },
-    async filter(
-      this: AspectResolverSubscriptions,
-      payload: AspectCommentsMessageReceived,
-      variables: { aspectID: string },
-      context: { req: { user: AgentInfo } }
-    ) {
-      const agentInfo = context.req.user;
-      const logMsgPrefix = `[User (${agentInfo.email}) Aspect comments]`;
-      this.logger.verbose?.(
-        `${logMsgPrefix} Filtering event '${payload.eventID}'`,
-        LogContext.SUBSCRIPTIONS
-      );
+  @TypedSubscription<AspectCommentsMessageReceived, AspectMessageReceivedArgs>(
+    () => AspectCommentsMessageReceived,
+    {
+      description: 'Receive new comment on Aspect',
+      resolve(this: AspectResolverSubscriptions, payload, args, context) {
+        const agentInfo = context.req?.user;
+        const logMsgPrefix = `[User (${agentInfo.email}) Aspect comments] - `;
+        this.logger.verbose?.(
+          `${logMsgPrefix} sending out new message event for Aspect: ${payload.aspectID} `,
+          LogContext.SUBSCRIPTIONS
+        );
+        return payload;
+      },
+      async filter(
+        this: AspectResolverSubscriptions,
+        payload,
+        variables,
+        context
+      ) {
+        const agentInfo = context.req.user;
+        const logMsgPrefix = `[User (${agentInfo.email}) Aspect comments]`;
+        this.logger.verbose?.(
+          `${logMsgPrefix} Filtering event '${payload.eventID}'`,
+          LogContext.SUBSCRIPTIONS
+        );
 
-      const isSameCommentsInstance = payload.aspectID === variables.aspectID;
-      this.logger.verbose?.(
-        `${logMsgPrefix} Filter result is ${isSameCommentsInstance}`,
-        LogContext.SUBSCRIPTIONS
-      );
-      return isSameCommentsInstance;
-    },
-  })
+        const isSameCommentsInstance = payload.aspectID === variables.aspectID;
+        this.logger.verbose?.(
+          `${logMsgPrefix} Filter result is ${isSameCommentsInstance}`,
+          LogContext.SUBSCRIPTIONS
+        );
+        return isSameCommentsInstance;
+      },
+    }
+  )
   async aspectCommentsMessageReceived(
     @CurrentUser() agentInfo: AgentInfo,
-    @Args({
-      name: 'aspectID',
-      type: () => UUID,
-      description: 'The ID of the Aspect to subscribe to.',
-      nullable: false,
-    })
-    aspectID: string
+    @Args({ nullable: false }) args: AspectMessageReceivedArgs
```

Comment:
The type name here needs consideration. I do not like args. AspectMessageReceivedSubscriptionArgs? Or we switch to using an input DTO. In the case of every mutation we have a DTO to wrap all arguments.

Reply:
Not an issue for this PR, we need a broader discussion on the args to use for Subscriptions